releasenotes/notes/k8s-cluster-name-format.yaml

```diff
@@ -0,0 +1,25 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+releaseNotes:
+  - |
+    **Added** For multicluster installations, istioctl has historically required
+    that a cluster name be a valid `DNS_LABEL` as defined by
+    [RFC 1123](https://tools.ietf.org/html/rfc1123).
+    This was due to the fact that cluster names are embedded in the name
+    of the generated remote secret and the most strict
+    [names in Kubernetes](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/identifiers.md)
+    have this requirement. Kubernetes object names, however, conform to
+    RFC 1123 `DNS_SUBDOMAIN`, which is more flexible both in length
+    (up to 253 characters) as well allowing the `.` character within the
+    string. Since the name of a remote secret falls under this category,
+    we have chosen to relax the requirements accordingly.
+    Cluster names can now be up to 233 characters, which is the max
```

Comment:
if you expect hostnames of length greater then 64 characters on Linux, I think you are expecting trouble. It (was) very difficult to change this constant in kernel.org, and as such DNS is a bit wacky in kernel.org. I am not sure how DNS_SUBDOMAINs are handled in Linux, but I would encourage a little research on the limits.